### PR TITLE
Rename project and rewrite specs

### PR DESCRIPTION
**Pull request by niksirbi**
As discussed in our meeting today, I renamed the project and website to`SWC-Blueprint`.
I also extensively rewrote the specification text, and tried to add more visual separation (e.g. with bullet points and boxes).
You should build the website locally to better judge the visual result (preview in the attached image).

@JoeZiminski I tried to have the specification match the current `datashuttle` features, but I may be mistaken in some places, so please take a look to confirm.

@adamltyson now the build is sent to `swc-blueprint.neuroinformatics.dev`. I don't know whether you need to change something on the hosting end for that to work. 

Screenshot 2023-03-10 at 21 47 31

### CODE REVIEW

**Comment by adamltyson** [comment]
> @adamltyson now the build is sent to swc-blueprint.neuroinformatics.dev. I don't know whether you need to change something on the hosting end for that to work.

@niksirbi I do, just let me know when this gets merged and I'll update.

File: docs/source/specification.md [hunk 1]
```diff
@@ -22,102 +27,147 @@ For example:
...
 ```
 
-The project folder may have any name descriptive of the project (that does not include spaces). Within the 
-project folder, data is separated into `rawdata` and `derivatives`. `rawdata` contains the raw
-data acquired during data collection. `derivatives` includes any processed data (e.g. spike sorting or pose estimation). 
-The subject / session / datatype folder structure within derivatives should
-match the rawdata folder whenever possible.
+### Basic principles
+
+* The project folder ***may*** have any name descriptive of the project but this ***must not*** contain spaces.
```
Comment (JoeZiminski) [inline docs/source/specification.md:32]: For me personally, I prefer keywords of required, optional and recommended in italics only. I find the words must / must not with bold a little commanding / annoying. This mostly comes from my first memory of reading BIDS, thinking that it is already a pain that I am going to re-organise my whole directory structure and they are going to say I MUST? do xyz? But, maybe most people don't mind and I have some unresolved issues with being told what to do 😅

File: docs/source/specification.md [hunk 2]
```diff
@@ -22,102 +27,147 @@ For example:
...
+
+  * `rawdata`: data coming out of the data acquisition system (e.g. binary files, tiffs, videos files). 
+
+  * `derivatives`: any processed data that is derived from `rawdata` (e.g. spike sorting or pose estimation).
 
-Each subject (e.g. mouse, rat) in the project has a dedicated folder in which all data from 
-experimental sessions are stored.
-In the case of multiple sessions, each session has its own folder where acquired data, belonging to
-a dataype (e.g. `ephys, behav, funcimg, histology`) are stored.
+* Data within the `rawdata` folder ***must*** be hierarchically structured into `subject/session/datatype` levels. Each level ***must*** contain at least one folder corresponding to the next (lower) level.
```
Comment (JoeZiminski) [inline docs/source/specification.md:40]: maybe an example here would be useful off the bat (e.g. `sub-001/ses-001/ephys`). [oh I forgot there is an example dir tree at the top, maybe this is not required]

File: docs/source/specification.md [hunk 3]
```diff
@@ -22,102 +27,147 @@ For example:
...
 
-Subject and session folder names cannot contain spaces and must consist of key-value pairs separated
-by underscores, e.g. `sub-001_id-5645332`. 
+* The `subject/session/datatype` folder structure within `derivatives` ***should***
```
Comment (JoeZiminski) [inline docs/source/specification.md:42]: Maybe discussion of derivative formatting can come at the end once the sub/ses/data_type hierarchy is fully introduced.

File: docs/source/specification.md [hunk 4]
```diff
@@ -22,102 +27,147 @@ For example:
...
+:::
 
 ### Session
-Within the subject directory reside session subdirectories named with the *required* prefix "ses", e.g. `ses-<label>`. A session represents a recording, 
-functional imaging or behavioural experimental session, and may contain multiple blocks, or runs, of data acquisition.
-Different sessions may have different combinations of datatypes. *optional* key - value pairs can be added after ses-<label>
-(e.g. `ses-001_date-220516`).
+
+* Session-level folders ***must*** be prefixed with a key-value pair that is unique for each session. The key ***must*** be "ses" and the value ***must*** be numerical, e.g. `ses-01`. 
+* It is ***recommended*** that sessions are assigned ascending numerical labels as they are added to the project. The labels ***should*** be prefixed with an arbitrary number of 0s for consistent indentation, e.g. `ses-01`, `ses-02`, `ses-03`.
```
Comment (JoeZiminski) [inline docs/source/specification.md:68]: not 100% tied to required, recommended, optional as in some instances they do make thing a bit worried e.g. `a must do b` vs. `a is required to do b`. However I think we should pick 3 only as avoids having to remember what categories each word belongs to (for example recommended vs. should vs. may do not have obvious groupings to me).

Comment (JoeZiminski) [inline docs/source/specification.md:68]: I really like this, I think the issue of what leading zeros are allowed is never properly explained

File: docs/source/specification.md [hunk 5]
```diff
@@ -22,102 +27,147 @@ For example:
...
+            │       └── sub-001_ses-01_tracking-results.csv
             └── histology/
-                └── sub-001_ses-001_cell_counts.csv
+                └── sub-001_ses-01_cell-counts.csv
 ```
 
 ## File Naming Conventions
 
-File names are *recommended* to be formatted as set of key-value pairs ending with an *optional* suffix and *required* file extension (e.g. `sub-001_ses-001_date-20220516_retinotopy.ap.bin`). 
+`SWC-Bluprint` does not impose any absolute requirements on file names. That said, below we provide some ***recommended*** and ***optional*** guidelines for file names, based on the [BIDS specification](https://bids-specification.readthedocs.io/en/stable/02-common-principles.html#filenames).
```
Comment (JoeZiminski) [inline docs/source/specification.md:121]: maybe just mention we provide recommended in this line? (we do also provide optional but key thing is for them to realise we recommending quite strongly a way they do it)

File: docs/source/specification.md [hunk 6]
```diff
@@ -22,102 +27,147 @@ For example:
...
+
+:::{admonition} What makes a good file name?
+:class: tip
+* be nice to humans (readable and descriptive)
```
Comment (JoeZiminski) [inline docs/source/specification.md:125]: awesome section

File: docs/source/specification.md [hunk 7]
```diff
@@ -22,102 +27,147 @@ For example:
...
+* be nice to computers (parseable and consistent)
+* use alphanumeric characters `Aa-Zz, 0-9`, dashes `-`, underscores `_`
```
Comment (JoeZiminski) [inline docs/source/specification.md:127]: could mention key-value pair arrangement is  preferred (maybe in the nice to computer line?)

**Review by JoeZiminski — COMMENTED** [review]
Great stuff @niksirbi I think this is a major improvement, much cleaner and easy to follow!

My only real suggestion is to use only three words to represent the three "levels", e.g. 'required, recommended and optional' (though not tied to these three specifically, discussed in comments). I think having multiple words per level as BIDS does reads quite confusingly (for me at least) as it is not always intuitive what category each word belongs do, requiring having to keep going back to the key and look things up.

File: docs/source/index.md [hunk 8]
```diff
@@ -0,0 +1,17 @@
+# SWC-Blueprint
+
+`SWC-Blueprint` is a project folder structure specification for use in the [Sainsbury Wellcome Centre (SWC) for Neural Circuits and Behaviour](https://www.sainsburywellcome.org/).
```
[the remark this answers is not in the document]
Comment (niksirbi) [reply docs/source/index.md:3]: I think  "folder" is more accessible to non-techy users

[the remark this answers is not in the document]
Comment (niksirbi) [reply docs/source/index.md:3]: I had checked throughout the specification, and it's always referred to as "foder", so it's at least consistent within this repo. Remains to be changed on datashuttle

File: docs/source/specification.md [hunk 9]
```diff
@@ -1,11 +1,16 @@
-# Specification
+# The specification
 
-## Project Folder Structure 
+:::{note}
+We mark specifications with ***keywords*** that should be interpreted as described in [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt). In decreasing order of requirement, these are:
```
[the remark this answers is not in the document]
Comment (niksirbi) [reply docs/source/specification.md:4]: It actually means "Request For Comments".
I know replaced it with "Network Working Group", which is the author of that document and may serve as a more meaningful indicator

File: docs/source/specification.md [hunk 10]
```diff
@@ -1,11 +1,16 @@
...
+
+* ***required, must, must not, shall, shall not***
```
[the remark this answers is not in the document]
Comment (niksirbi) [reply docs/source/specification.md]: Now I only use the three verbs (italicised): *must*, *should*, *may*. Hopefully, that's much less to keep in the reader's working memory.

File: docs/source/specification.md (hunk 1 above)
Comment (niksirbi) [reply docs/source/specification.md]: It's now italics

File: docs/source/specification.md (hunk 3 above)
Comment (niksirbi) [reply docs/source/specification.md]: Moved to after the bullet points

File: docs/source/specification.md (hunk 7 above)
Comment (niksirbi) [reply docs/source/specification.md:120]: I don't want to blow up this section with too many examples (want to keep the "tip" very general). Anyhow, all of our recommendations in the below bullet points are essentially about the key-value pair convention.

**Comment by JoeZiminski** [comment]
Notes from feedback:

- [ ] "sub" convert to `sub`
- [ ] explain why wrong examples are wrong
- [ ] "e.g." is missing on behaviour example
- [ ] change "calcium imaging and voltage imaging" to "calcium and voltage imaging"
- [ ] "SWC-Bluprint" is typo
- [ ] remove "suffix" option for now. There are also suffixes in some examples that can be removed
- [ ] remove camelcase example
- [ ] "af.bin" file extension in example is wrong (can change to .ap or .lf)

**Comment by niksirbi** [comment]
- [ ] Clarify that we did not forget about metadata

**Comment by niksirbi** [comment]
Hey @JoeZiminski, I have applied the changes suggested by the feedback, and also added a small metadata section, recommending the use of TSV files for tabular metadata. 
Feel free to proofread one last time, approve and merge.

**Review by JoeZiminski — APPROVED** [review]

**Pull request merged**